### PR TITLE
Remove deleted trajectories from submaps display. 

### PR DESCRIPTION
**Pull request by MichaelGrupp**
Fixes strange behavior with empty leftover entries in the trajectory list of the RViz plugin.

By using an `std::map` instead of an `std::vector` for the trajectories  we can
easily erase trajectories which do not appear in the submap list anymore.

### CODE REVIEW

**Comment by MichaelGrupp** [comment]
### Before:
screenshot from 2018-10-16 14-35-29

### After:
screenshot from 2018-10-16 14-34-50

File: cartographer_rviz/cartographer_rviz/submaps_display.cc [hunk 1]
```diff
@@ -172,10 +177,18 @@ void SubmapsDisplay::processMessage(
...
+      trajectories_.erase(it++);
+    } else {
+      ++it;
+    }
+  }
   // Remove all submaps not mentioned in the SubmapList.
-  for (size_t trajectory_id = 0; trajectory_id < trajectories_.size();
-       ++trajectory_id) {
-    auto& trajectory_submaps = trajectories_[trajectory_id]->submaps;
+  for (const auto& trajectory_by_id : trajectories_) {
```
Comment (MichaelGrupp) [inline cartographer_rviz/cartographer_rviz/submaps_display.cc]: if you have a better name for this `std::pair<int, cartographer_rviz::Trajectory>` than "trajectory_by_id` - let me know

File: cartographer_rviz/cartographer_rviz/submaps_display.cc [hunk 2]
```diff
@@ -172,10 +177,18 @@ void SubmapsDisplay::processMessage(
     }
     trajectory_submaps.at(id.submap_index)->Update(msg->header, submap_entry);
   }
+  // Remove all deleted trajectories not mentioned in the SubmapList.
+  for (auto it = trajectories_.begin(); it != trajectories_.end();) {
+    if (listed_trajectories.count(it->first) == 0) {
+      trajectories_.erase(it++);
```
Comment (gaschler) [inline cartographer_rviz/cartographer_rviz/submaps_display.cc]: it = erase(it)

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**